jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/xloader/ProcIndexBuildX.java

```diff
@@ -143,31 +143,49 @@ private static long sort_build_index(Logger LOG, String datafile, DatasetGraph d
         Process proc2;
         OutputStream toSortOutputStream; // Not used. Input is a file.
         InputStream fromSortInputStream;
+
         try {
             //LOG.info("Step : external sort : "+indexName);
             //if ( sortArgs != null ) {}
+
+
             List<String> sortCmdBasics = Arrays.asList(
                  "sort",
                     "--temporary-directory="+TMPDIR, "--buffer-size=50%",
                     "--parallel=2", "--unique"
                     //, "--compress-program=/usr/bin/gzip"
-                );
+            );
             List<String> sortCmd = new ArrayList<>(sortCmdBasics);
+            if ( BulkLoaderX.CompressSortIndexFiles )
+                sortCmd.add("--compress-program=/usr/bin/gzip");
```

Comment:
Should this be configurable in case this is used on some system with a weird installation layout?  Or better still can this just be `gzip`?

Reply:
Where do do programs live on macOS?

I'm not sure that the fork-exec in sort(1) will respect the PATH. Java ProcessBuilder does - so "sort" is OK.

Reply:
It lives in the same place, just having spent a lot of time in containers often using minimal Linux distro's over the last few years I've become very wary of assuming things will a) exist and b) always be where you expect

So it looks like they use `execlp` - https://github.com/wertarbyte/coreutils/blob/master/src/sort.c#L1084 - and the Linux manual pages say the following:

> # Special semantics for execlp() and execvp()

> The execlp(), execvp(), and execvpe() functions duplicate the actions of the shell in searching for an executable file if the specified filename does not contain a slash (/) character. The file is sought in the colon-separated list of directory pathnames specified in the PATH environment variable. If this variable isn't defined, the path list defaults to the current directory followed by the list of directories returned by confstr(_CS_PATH). (This confstr(3) call typically returns the value "/bin:/usr/bin".)
If the specified filename includes a slash character, then PATH is ignored, and the file at the specified pathname is executed.

So you should be able to pass in `gzip` and have it search the `PATH` just fine